arch/WASM/WASMMapping.c

```diff
@@ -0,0 +1,333 @@
+	{ WASM_INS_I64_TRUNC_U_F64, "f64.trunc_u/f64" },
+	{ WASM_INS_F32_CONVERT_S_I32, "f32.convert_s/i32" },
+	{ WASM_INS_F32_CONVERT_U_I32, "f32.convert_u/i32" },
+	{ WASM_INS_F32_CONVERT_S_I64, "f32.convert_s/i64" },
+	{ WASM_INS_F32_CONVERT_U_I64, "f32.convert_u/i64" },
+	{ WASM_INS_F32_DEMOTE_F64, "f32.demote/f64" },
+	{ WASM_INS_F64_CONVERT_S_I32, "f64.convert_s/i32" },
+	{ WASM_INS_F64_CONVERT_U_I32, "f64.convert_u/i32" },
+	{ WASM_INS_F64_CONVERT_S_I64, "f64.convert_s/i64" },
+	{ WASM_INS_F64_CONVERT_U_I64, "f64.convert_u/i64" },
+	{ WASM_INS_F64_PROMOTE_F32, "f64.promote/f32" },
+	{ WASM_INS_I32_REINTERPRET_F32, "i32.reinterpret/f32" },
+	{ WASM_INS_I64_REINTERPRET_F64, "i64.reinterpret/f64" },
+	{ WASM_INS_F32_REINTERPRET_I32, "f32.reinterpret/i32" },
+	{ WASM_INS_F64_REINTERPRET_I64, "f64.reinterpret/i64" },
+	{ WASM_INS_INVALID, NULL },
+	{ WASM_INS_INVALID, NULL },
+	{ WASM_INS_INVALID, NULL },
+	{ WASM_INS_INVALID, NULL },
+	{ WASM_INS_INVALID, NULL },
+	{ WASM_INS_INVALID, NULL },
+	{ WASM_INS_INVALID, NULL },
+	{ WASM_INS_INVALID, NULL },
+	{ WASM_INS_INVALID, NULL },
+	{ WASM_INS_INVALID, NULL },
+	{ WASM_INS_INVALID, NULL },
+	{ WASM_INS_INVALID, NULL },
+	{ WASM_INS_INVALID, NULL },
+	{ WASM_INS_INVALID, NULL },
+	{ WASM_INS_INVALID, NULL },
+	{ WASM_INS_INVALID, NULL },
+	{ WASM_INS_INVALID, NULL },
+	{ WASM_INS_INVALID, NULL },
+	{ WASM_INS_INVALID, NULL },
+	{ WASM_INS_INVALID, NULL },
+	{ WASM_INS_INVALID, NULL },
+	{ WASM_INS_INVALID, NULL },
+	{ WASM_INS_INVALID, NULL },
+	{ WASM_INS_INVALID, NULL },
+	{ WASM_INS_INVALID, NULL },
+	{ WASM_INS_INVALID, NULL },
+	{ WASM_INS_INVALID, NULL },
+	{ WASM_INS_INVALID, NULL },
+	{ WASM_INS_INVALID, NULL },
+	{ WASM_INS_INVALID, NULL },
+	{ WASM_INS_INVALID, NULL },
+	{ WASM_INS_INVALID, NULL },
+	{ WASM_INS_INVALID, NULL },
+	{ WASM_INS_INVALID, NULL },
+	{ WASM_INS_INVALID, NULL },
+	{ WASM_INS_INVALID, NULL },
+	{ WASM_INS_INVALID, NULL },
+	{ WASM_INS_INVALID, NULL },
+	{ WASM_INS_INVALID, NULL },
+	{ WASM_INS_INVALID, NULL },
+	{ WASM_INS_INVALID, NULL },
+	{ WASM_INS_INVALID, NULL },
+	{ WASM_INS_INVALID, NULL },
+	{ WASM_INS_INVALID, NULL },
+	{ WASM_INS_INVALID, NULL },
+	{ WASM_INS_INVALID, NULL },
+	{ WASM_INS_INVALID, NULL },
+	{ WASM_INS_INVALID, NULL },
+	{ WASM_INS_INVALID, NULL },
+	{ WASM_INS_INVALID, NULL },
+	{ WASM_INS_INVALID, NULL },
+	{ WASM_INS_INVALID, NULL },
+	{ WASM_INS_INVALID, NULL },
+	{ WASM_INS_INVALID, NULL },
+	{ WASM_INS_INVALID, NULL },
+	{ WASM_INS_INVALID, NULL },
+	{ WASM_INS_INVALID, NULL },
+	{ WASM_INS_INVALID, NULL },
+	{ WASM_INS_INVALID, NULL },
+	{ WASM_INS_INVALID, NULL },
+	{ WASM_INS_INVALID, NULL },
+	{ WASM_INS_INVALID, NULL },
+	{ WASM_INS_INVALID, NULL },
+	{ WASM_INS_INVALID, NULL },
+};
+#endif
+
+const char *WASM_insn_name(csh handle, unsigned int id)
+{
+#ifndef CAPSTONE_DIET
+	if (id >= ARR_SIZE(insn_name_maps))
+		return NULL;
+	else
+		return insn_name_maps[id].name;
+#else
+	return NULL;
```

Comment:
in diet mode wasm disassembler not work

Reply:
@SpikeI can you take a look at Diet mode?